server/src/main/java/org/opensearch/index/store/RemoteIndexInput.java

```diff
@@ -61,22 +71,25 @@ public long length() {
         return size;
     }
 
-    @Override
-    public void seek(long pos) throws IOException {
-        inputStream.skip(pos);
-    }
-
     /**
      * Guaranteed to throw an exception and leave the RemoteIndexInput unmodified.
      * This method is not implemented as it is not used for the file transfer to/from the remote store.
      *
      * @throws UnsupportedOperationException always
      */
     @Override
-    public long getFilePointer() {
+    public void seek(long pos) throws IOException {
```

Comment:
Just curious, why `inputStream.skip(pos);` was replaced with `UnsupportedOperationException` ?

Reply:
The implementation was incorrect. This was pointed by @andrross earlier. seek() should provide functionality to go to the exact point provided by the `pos` argument. That means `pos` could be 0 and it will take you to the beginning of the inputStream. This requires special handling of the state which is not required at least for the V1 of remote store.